pkg/api/dirs_test.go

```diff
@@ -274,62 +271,18 @@ Disallow: /`),
 			},
 		},
 	} {
-		t.Run(tc.name, func(t *testing.T) {
-			// tar all the test case files
-			tarReader := tarFiles(t, tc.files)
-
-			var resp api.FileUploadResponse
-
-			options := []jsonhttptest.Option{
-				jsonhttptest.WithRequestBody(tarReader),
-				jsonhttptest.WithRequestHeader("Content-Type", api.ContentTypeTar),
-				jsonhttptest.WithUnmarshalJSONResponse(&resp),
-			}
-			if tc.indexFilenameOption != nil {
-				options = append(options, tc.indexFilenameOption)
-			}
-			if tc.errorFilenameOption != nil {
-				options = append(options, tc.errorFilenameOption)
-			}
-			if tc.encrypt {
-				options = append(options, jsonhttptest.WithRequestHeader(api.SwarmEncryptHeader, "true"))
-			}
-
-			// verify directory tar upload response
-			jsonhttptest.Request(t, client, http.MethodPost, dirUploadResource, http.StatusOK, options...)
-
-			if resp.Reference.String() == "" {
-				t.Fatalf("expected file reference, did not got any")
-			}
-
+		verify := func(t *testing.T, resp api.FileUploadResponse) {
+			t.Helper()
 			// NOTE: reference will be different each time when encryption is enabled
-			if !tc.encrypt {
-				if !resp.Reference.Equal(tc.expectedReference) {
-					t.Fatalf("expected root reference to match %s, got %s", tc.expectedReference, resp.Reference)
-				}
-			}
-
-			// read manifest metadata
-			j, _, err := joiner.New(context.Background(), storer, resp.Reference)
-			if err != nil {
-				t.Fatal(err)
-			}
-
-			buf := bytes.NewBuffer(nil)
-			_, err = file.JoinReadAll(context.Background(), j, buf)
-			if err != nil {
-				t.Fatal(err)
-			}
-			e := &entry.Entry{}
-			err = e.UnmarshalBinary(buf.Bytes())
-			if err != nil {
-				t.Fatal(err)
-			}
+			// if !tc.encrypt {
```

Comment:
do we want to remove this? if yes - why? if not - can we make it work? :)

Reply:
It works actually... I had commented during some changes.